opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -328,33 +328,46 @@ class SimulatorFullyImplicitBlackoilEbos
             std::vector<std::vector<double>> currentFluidInPlace;
             currentFluidInPlace = solver->computeFluidInPlace(fipnum);
             std::vector<double> currentFluidInPlaceTotals = FIPTotals(currentFluidInPlace, state);
+			      const std::string version = moduleVersionName();
 
             FIPUnitConvert(eclState().getUnits(), currentFluidInPlace);
             FIPUnitConvert(eclState().getUnits(), currentFluidInPlaceTotals);
 
+            if ( output_writer_.output() ) {
+                if ( output_writer_.isIORank() )
+                {
+                    stepReport.reportParam(tstep_os);
+                }
+            }
+
+            // Increment timer, remember well state.
+            ++timer;
+			
             if (terminal_output_ )
             {
+                std::ostringstream ss;
```

Comment:
Please put the following code in a separate function (the run() function is far too long as it is...)